### PR TITLE
Merge of Bigbrain Viewer capability into main branch.

### PR DESCRIPTION
**Pull request by natacha-beck**
In this PR I tried to merge everything from Lindsay work on the [BigBrain viewer](http://aces.github.io/bigbrain-surface-viewer/) the code source can be found [here](https://github.com/aces/bigbrain-surface-viewer/). 

For the moment you will found the bigbrain viewer under examples/bigbrain-viewer, in this way you can test the functionality.

I tried to generalize the functionality added by Lindsay and to put this one in appropriate files (see `viewer.gridHelper` for example). I also fixed some little stuffs in same time.  

Here is some notes about what I have done during the merge process: 

> 
> Integration of BBl:
> 
> Step 1 (27c91bfa7fbebe6362124a487452c11d751f0fe0):
>   - Take the surface-viewer-demo.js in https://github.com/aces/bigbrain-surface-viewer put in examples/bigbrain-viewer-demo.js
>   - Take the index.html             in https://github.com/aces/bigbrain-surface-viewer put in examples/bigbrain-viewer-demo.html
> 
> Step 2 (2bf03ee260c05e06df03d604f2100b30f40648f2):
>   - Added missing images
> 
> Step 3 (186596028e5c5a3651a9281b2c1c72439e17981b):
>   - Added missing css, and *.js file for the bigbrain viewer + fix path in code.
> 
> Step 4 (2fe2ca2ab3b013f8f158f8c39961ef3af257fb6b):
>   - Run jshint + fix some of the warning (indentation + equality operator)
> 
> Step 5 (8fb11bc2e5461cb9a6bbf979fe293b66b1f2012c):
>   - More jshint fix.
>   - Added Bigbrain Grid in Three.js                        (TODO: should not be here)
>   - In rendering.js add some code to handle axes (TODO: should not be here)
> 
> Step 6 (1a8e11df4a096b2b53e1d5efcd18a5a927e96634):
>   - more fix for jshint (axes)
> 
> Step 7 (8a06fa89cd74b6c74171c7c4d1395633b56aab6f):
>   - Replace more for loop by forEach loop.
> 
> Step 8 (61b5009579c99ffb76fce9c686569f84b826062a):
>   - Add jshint rules to ignore loopfunc and eval usage
>   - Fixed usage of searchapes as a global variable.
>   - Temporary added getVertex2 in src/brainbrowser/surface-viewer.js (TODO: should not be here)
>   - Temporary added a version of viewer.pick_bb in src/brainbrowser/surface-viewer/modules/rendering.js (TODO: should not be here)
> 
> Step 9 (7cb489b6dd92afd4a0f147daa47665fc0a10c4bd):
>   - Some more fix for jshint
> 
> Step 10 (28ed43e00d48b106f64b8162ea7bb05ea14f62ea):
>   - Fix of Grid and fix of getVertex2
> 
> Step 11 (17ba618d014693b6b5ef805dfee1d999db8e13ae):
>   - Split pick_bb in 2: 1 == the old one, 2 == pick by index.
>   - Remove getVertex2
> 
> 
> Step 12 (f287c9c4d619614b39ed84050cb7b3241f2ec603,
>          bee7171fbd940a489efcebb52135474c181975e0,
>          0a3eb5e84c0f0d985ec834fbe50f81f26ac28db5,
>          99a8ad86cc64c2d66ea5fd8a9bf6b8aa9369c926 )
>   - Extract gridHelper.
> 
> Step 13 (7568f60e5ee6d10bce96b79daa75f28b6ce66b54):
>   - Fix rotation of related models
> 
> Step 14 (1bf02a0c257b53b4cc3f32827f8262c1cd9e92d4):
>   - Fix bug with color slider when no color available ==> provide a default
> 
> Step 15 (49be218212952b45da63aa3e019b350007bdcd85):
>   - Fix format and autoload issues

Here is some general comment/issue that seems to be present: 
  - This version cannot load *.gz files for the moment (==> with the new version off BB this should be possible)
 - I'm not sure of the comportment of BB when a user load two files with model that have the same names. Seems to be ok. 
 - the viewer.gridHelper need more documentation but it's not clear for me for the moment (that is why you can see some `???`)

Please let me know all yours comments, I'm sure this PR have place for improvement.

### CODE REVIEW

File: src/brainbrowser/surface-viewer/modules/rendering.js [hunk 1]
```diff
@@ -631,6 +688,62 @@ BrainBrowser.SurfaceViewer.modules.rendering = function(viewer) {
...
+  * ```js
+  * viewer.pickByVertex(2356);
+  * viewer.pickByVertex(2356, { model_name: "brain.obj" });
+  * ```
+  */
+  viewer.pickByVertex = function(index, options) {
+    var model = viewer.model;
+    options   = options || {};
+
+    if ((index !== "") && !(/^\d+$/.test(index))) {return null;}
```
Comment (rdvincent) [inline src/brainbrowser/surface-viewer/modules/rendering.js]: Is index a number or a string?

Comment (natacha-beck) [inline src/brainbrowser/surface-viewer/modules/rendering.js]: It should be a integer at this point. not a string, thanks for the comment.

**Comment by natacha-beck** [comment]
Here is some issue reported: 

- Per vertex data on FS ASC surface does not work (blank map)
- `Grid length`conflict with auto resize.
- Wireframe button seems to be buggy (act different on brainbrowser and bigbrain).
- buildGrid is called when slider is used even if grid is not set.
- Having grid unable => Search for shape => clear selection ==  weird effect the grid is not resized. 
- The viewer failed to load compressed file.

File: src/brainbrowser/surface-viewer/modules/rendering.js (hunk 1 above)
Comment (natacha-beck) [inline src/brainbrowser/surface-viewer/modules/rendering.js]: I modified the code to be more coherent here.

**Comment by natacha-beck** [comment]
At this point all the previous comments are fixed. Please take a look and let me know if you find something.

**Comment by natacha-beck** [comment]
New batch of comment from @lblewis : 

-Little typo for filetype dropdown menu, "Unknow" should be "Unknown"
-When I initially load in the .json, it is no longer rotating around the center of the .json!!
-If grid / axes are on, and I then shift-click a shape, and I click on Focus, and then on Unfocus, all the grid / axes / legend disappear (they should not)
-If I do the same as above, but this time I set Grid auto-resize toggle to Off before I click on Focus, then when I click on Unfocus, the axes and legend disappears (but not the grid)

**Comment by natacha-beck** [comment]
Need to fix this one from @lblewis 

- When I initially load in the .json, it is no longer rotating around the center of the .json!! --> should be done, not yet the case on this version.

**Comment by natacha-beck** [comment]
Fix the 'model centric' part.

**Comment by natacha-beck** [comment]
Just fix the last bug that @lblewis had mention. Please go head for a code review.

**Comment by natacha-beck** [comment]
With @jonathanlurie we have observed some strange behaviour concerning the grid. When a user manipulates the opacity of the shape, then the grid resizes, but the center of rotation stays the same; this kind of behaviour can thus generate some strange effects (this comportement is insync with @lblewis version).

File: src/brainbrowser/surface-viewer/modules/views.js [hunk 2]
```diff
@@ -306,8 +307,8 @@ BrainBrowser.SurfaceViewer.modules.views = function(viewer) {
     active_wireframe_jobs++;
   }
 
-  function toggleWireframe(shape, wireframe, is_wireframe) {
-    shape.material.visible = !is_wireframe;
+  function toggleWireframe(shape, wireframe, is_wireframe, keep_surface) {
+    shape.material.visible = keep_surface === true ? true : !is_wireframe;
```
Comment (rdvincent) [inline src/brainbrowser/surface-viewer/modules/views.js]: shape.material.visible = keep_surface || !is_wireframe

File: src/brainbrowser/surface-viewer/modules/rendering.js [hunk 3]
```diff
@@ -519,6 +575,7 @@ BrainBrowser.SurfaceViewer.modules.rendering = function(viewer) {
     intersects = raycaster.intersectObject(model, true);
 
     for (i = 0; i < intersects.length; i++) {
+      intersects[i].object.userData.pick_ignore = (intersects[i].object.material.opacity < opacity_threshold) ? true : false;
```
Comment (rdvincent) [inline src/brainbrowser/surface-viewer/modules/rendering.js]: intersects[i].object.userData.pick_ignore = (intersects[i].object.material.opacity < opacity_threshold);

File: src/brainbrowser/surface-viewer/modules/views.js (hunk 2 above)
Comment (natacha-beck) [inline src/brainbrowser/surface-viewer/modules/views.js]: Done

File: src/brainbrowser/surface-viewer/modules/rendering.js (hunk 3 above)
Comment (natacha-beck) [inline src/brainbrowser/surface-viewer/modules/rendering.js]: done

**Comment by natacha-beck** [comment]
@rdvincent I just done the update, with your comments.

**Pull request merged**